src/main/java/core/basesyntax/dataprocess/DataProcessor.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.dataprocess;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.Strategy;
+import core.basesyntax.strategy.impl.BalanceService;
+import core.basesyntax.strategy.impl.PurchaseService;
+import core.basesyntax.strategy.impl.ReturnService;
+import core.basesyntax.strategy.impl.SupplyService;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DataProcessor {
+    private final List<FruitTransaction> fruitTransactions;
+    private final Map<String, Integer> fruitData;
+    private final Map<String, Strategy> strategyMap;
+
+    public DataProcessor(List<FruitTransaction> fruitTransactions, Map<String, Integer> fruitData) {
+        this.fruitTransactions = fruitTransactions;
+        this.fruitData = fruitData;
+        this.strategyMap = initializeStrategyMap();
```

Comment:
pass map as an argument to constructor for greater flexibility
init it in main via map.of() for concise code